dashboard/dashboard/views.py

```diff
@@ -125,6 +126,15 @@ def view_administration(request):
         context['plugins'] = plugin_client.get_plugins_info()
         context['job_type_versions'] = plugin_client.get_job_type_versions()
         context['infrastructure_targets'] = plugin_client.get_infrastructure_targets()
+
+        #Collect instances running on the same infrastructure
+        _infrastructure_instances: dict = collections.defaultdict(list)
+        for instance, infrastructure in context['infrastructure_targets'].items():
+            _versioned_name: str = infrastructure['name'] + ' (' + infrastructure['version'] + ')'
```

Comment:
I believe we can just use the `infrastructure['name']` (or `plugin_manifest['name']`) as a key. If we have 2 infrastructures provided by different versions of plugins, we can group them together.